distributed/nanny.py

```diff
@@ -863,7 +863,7 @@ def watch_stop_q():
                 """
                 while True:
                     try:
-                        msg = child_stop_q.get(timeout=1000)
+                        msg = child_stop_q.get(timeout=1)
```

Comment:
I think this while True around a large timeout is intentional and working around a python 2 bug https://bugs.python.org/issue8844

If so, I think it should just wait forever instead
```suggestion
                        msg = child_stop_q.get()
```

Reply:
Happy to give it a shot